src/control/__tests__/createIndex.validation.test.ts

```diff
@@ -79,13 +108,128 @@ describe('createIndex argument validations', () => {
 
     test('should throw if dimension is not a positive integer', async () => {
       const toThrow = async () =>
-        await createIndex(IOA)({ name: 'index-name', dimension: -10 });
+        await createIndex(IOA)({
+          name: 'index-name',
+          dimension: -10,
+          region: 'us-east3',
+          cloud: 'gcp',
+          capacityMode: 'pod',
+        });
 
       expect(toThrow).rejects.toThrowError(PineconeArgumentError);
       expect(toThrow).rejects.toThrowError(
         "The argument to createIndex had validation errors: property 'dimension' must be >= 1."
       );
     });
+
+    test('should throw if region is not provided', () => {
+      const toThrow = async () =>
+        await createIndex(IOA)({
+          name: 'index-name',
+          dimension: 10,
+          cloud: 'gcp',
+          capacityMode: 'pod',
+        });
+
+      expect(toThrow).rejects.toThrowError(PineconeArgumentError);
+      expect(toThrow).rejects.toThrowError(
+        'The argument to createIndex must have required property: region.'
+      );
+    });
+
+    test('should throw if region is not a string', () => {
+      const toThrow = async () =>
+        await createIndex(IOA)({
+          name: 'index-name',
+          dimension: 10,
+          region: 111,
+          cloud: 'gcp',
+          capacityMode: 'pod',
+        });
+
+      expect(toThrow).rejects.toThrowError(PineconeArgumentError);
+      expect(toThrow).rejects.toThrowError(
+        "The argument to createIndex had type errors: property 'region' must be string."
+      );
+    });
+
+    test('should throw if cloud is not provided', () => {
+      const toThrow = async () =>
+        await createIndex(IOA)({
+          name: 'index-name',
+          dimension: 10,
+          region: 'us-east3',
+          capacityMode: 'pod',
+        });
+
+      expect(toThrow).rejects.toThrowError(PineconeArgumentError);
+      expect(toThrow).rejects.toThrowError(
+        'The argument to createIndex must have required property: cloud.'
+      );
+    });
+
+    test('should throw if cloud is not a string', () => {
+      const toThrow = async () =>
+        await createIndex(IOA)({
+          name: 'index-name',
+          dimension: 10,
+          region: 'us-east3',
+          cloud: 123,
+          capacityMode: 'pod',
+        });
+
+      expect(toThrow).rejects.toThrowError(PineconeArgumentError);
+      expect(toThrow).rejects.toThrowError(
+        'The argument to createIndex accepts multiple types. Either 1) 2) 3)'
```

Comment:
This is the `errorFormatter` issue mentioned below (`/src/control/types.ts`): https://github.com/pinecone-io/pinecone-ts-client/pull/153/files#diff-cf36f0ee9dc687ce8159fbecd0992953a0c2e646eac44cca76cc2a2ba8246db3R30

We haven't used typebox literals anywhere else and the `errorFormatter` doesn't play nice.